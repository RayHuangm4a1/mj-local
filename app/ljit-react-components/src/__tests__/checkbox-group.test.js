import React from 'react';
import { shallow, mount, } from 'enzyme';
import CheckBoxGroup from '../components/checkbox-group';

jest.mock('antd/lib/checkbox');

describe('CheckBox Group', () => {
	it('handle default props', () => {
		const {
			disabled,
			className,
			onChange,
		} = CheckBoxGroup.defaultProps;

		expect(disabled).toEqual(false);
		expect(className).toEqual('');
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should be selectable by class ljit-checkbox-group', () => {
		const wrapper = shallow(<CheckBoxGroup />);

		expect(wrapper.hasClass('ljit-checkbox-group')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<CheckBoxGroup className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should renders correctly', () => {
		const value = ['option 1',];
		const options = [
			{ label: 'option 1', value: 'option 1', },
		];
		const wrapper = shallow(
			<CheckBoxGroup
				value={value}
				options={options}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const defaultValue = [];
		const value = ['option 1',];
		const name = 'CheckBoxGroup';
		const options = [
			{ label: 'option 1', value: 'option 1', },
			{ label: 'option 2', value: 'option 2', },
			{ label: 'option 3', value: 'option 3', },
		];
		const onChange = () => {};
		const disabled = false;
		const className = 'mock-calss';
		const wrapper = mount(
			<CheckBoxGroup
				defaultValue={defaultValue}
				value={value}
				name={name}
				options={options}
				onChange={onChange}
				disabled={disabled}
				className={className}
			/>
		);

		expect(wrapper.props().defaultValue).toEqual(defaultValue);
		expect(wrapper.props().value).toEqual(value);
		expect(wrapper.props().name).toEqual(name);
		expect(wrapper.props().options).toMatchObject(options);
		expect(wrapper.props().onChange).toEqual(onChange);
		expect(wrapper.props().disabled).toEqual(disabled);
		expect(wrapper.props().className).toEqual(className);
	});

	describe('when pass options props', () => {
		it('should renders three options', () => {
			const options = [
				{ label: 'option 1', value: 'option 1', },
				{ label: 'option 2', value: 'option 2', },
				{ label: 'option 3', value: 'option 3', },
			];
			const wrapper = shallow(<CheckBoxGroup options={options} />);

			expect(wrapper.find('.ljit-checkbox-group__option')).toHaveLength(3);
		});
	});
});
