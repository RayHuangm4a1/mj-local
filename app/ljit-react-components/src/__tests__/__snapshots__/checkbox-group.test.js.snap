// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Group should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBoxGroup
    className=""
    disabled={false}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "option 1",
          "value": "option 1",
        },
      ]
    }
    value={
      Array [
        "option 1",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Checkbox
          className="ljit-checkbox-group__option"
          indeterminate={false}
          value="option 1"
        >
          option 1
        </Checkbox>,
      ],
      "className": "ljit-checkbox-group",
      "defaultValue": undefined,
      "disabled": false,
      "name": undefined,
      "onChange": [Function],
      "options": Array [],
      "value": Array [
        "option 1",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "option 1",
        "nodeType": "class",
        "props": Object {
          "autoFocus": undefined,
          "checked": undefined,
          "children": "option 1",
          "className": "ljit-checkbox-group__option",
          "disabled": undefined,
          "indeterminate": false,
          "name": undefined,
          "value": "option 1",
        },
        "ref": null,
        "rendered": "option 1",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Checkbox
            className="ljit-checkbox-group__option"
            indeterminate={false}
            value="option 1"
          >
            option 1
          </Checkbox>,
        ],
        "className": "ljit-checkbox-group",
        "defaultValue": undefined,
        "disabled": false,
        "name": undefined,
        "onChange": [Function],
        "options": Array [],
        "value": Array [
          "option 1",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "option 1",
          "nodeType": "class",
          "props": Object {
            "autoFocus": undefined,
            "checked": undefined,
            "children": "option 1",
            "className": "ljit-checkbox-group__option",
            "disabled": undefined,
            "indeterminate": false,
            "name": undefined,
            "value": "option 1",
          },
          "ref": null,
          "rendered": "option 1",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
